ext/EvoTreesCUDAExt/fit.jl

```diff
@@ -1,267 +1,330 @@
+            @inbounds for k in 1:K_val
+                g = nodes_sum[k, node]
+                h = nodes_sum[K_val+k, node]
+                if w <= zero(w) || h + lambda * w / K_val <= zero(h)
+                    tree_pred[k, node] = 0.0f0
+                else
+                    tree_pred[k, node] = -g / (h + lambda * w / K_val + epsv)
                 end
             end
-            # find best split
-            best = findmax(view(nodes[1].gains, :, js))
-            best_gain = best[1]
-            best_bin = best[2][1]
-            best_feat = js[best[2][2]]
-            if best_gain > params.gamma
-                for n in n_current
-                    tree.gain[n] = best_gain
-                    tree.cond_bin[n] = best_bin
-                    tree.feat[n] = best_feat
-                    tree.split[n] = best_bin != 0
-
-                    _left, _right = split_set!(
-                        nodes[n].is,
-                        is,
-                        left,
-                        right,
-                        x_bin,
-                        tree.feat[n],
-                        tree.cond_bin[n],
-                        feattypes[best_feat],
-                        offset,
-                    )
-                    offset += length(nodes[n].is)
-
-                    nodes[n<<1].is, nodes[n<<1+1].is = _left, _right
-                    nodes[n<<1].∑ .= nodes[n].hL[:, best_bin, best_feat]
-                    nodes[n<<1+1].∑ .= nodes[n].hR[:, best_bin, best_feat]
+        end
+    end
+end
 
-                    if length(_right) >= length(_left)
-                        push!(n_next, n << 1)
-                        push!(n_next, n << 1 + 1)
-                    else
-                        push!(n_next, n << 1 + 1)
-                        push!(n_next, n << 1)
-                    end
-                end
-            else
-                for n in n_current
-                    if L <: Quantile
-                        pred_leaf_cpu!(tree.pred, n, nodes[n].∑, L, params, ∇, nodes[n].is)
-                    else
-                        pred_leaf_cpu!(tree.pred, n, nodes[n].∑, L, params)
-                    end
-                end
+@kernel function get_gain_gpu!(
+    nodes_gain::AbstractVector{T}, 
+    nodes_sum::AbstractArray{T,2}, 
+    nodes, 
+    lambda::T, 
+    K_val::Int
+) where {T}
+    n_idx = @index(Global)
+    node = nodes[n_idx]
+    
+    @inbounds if node > 0
+        eps = T(1e-8)
+        
+        if K_val == 1
+            g = nodes_sum[1, node]
+            h = nodes_sum[2, node]
+            w = nodes_sum[2*K_val+1, node]
+            nodes_gain[node] = g^2 / (h + lambda * w + eps)
+        else
+            gain_sum = zero(T)
+            w = nodes_sum[2*K_val+1, node]
+            
+            @inbounds for k in 1:K_val
+                g = nodes_sum[k, node]
+                h = nodes_sum[K_val+k, node]
+                gain_sum += g^2 / (h + lambda * w / K_val + eps)
```

Comment:
Same `/K_val` comment. 

Also, why not use `max(eps, ...)` like in `cpu` implementation?